### PR TITLE
Remove run configuration items

### PR DESCRIPTION
**Pull request by patricktnast**
### Description

- *Category*: 
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5314

### Changes and notes
 
Remove redundant "run_configuration" items from updated job parameters

Instead, add a "sim_config" propery for the simulation_context configuration, and just use the format of job_parameters.job_specific (or job_parameters.branch_configuration if we don't need the draw and seed)

removed places where redundant information was propagated or ignored; eyes from @stevebachmeier regarding the results processing and @hussain-jafari regarding the central performance logging would be helpful to make sure im not doing anything weird.
### Testing

Tested with /ihme/homes/pnast/scratch/vivarium/vivarium_gates_nutrition_optimization_child/backup_testing/pakistan/2024_09_11_14_46_24
in similar manner to https://github.com/ihmeuw/vivarium_cluster_tools/pull/233 ; ran one job to completion, ran 3 more jobs, killing partway after backup creation, observed loading from saved backup.

also adjusted some unit tests

### CODE REVIEW

**Comment by patricktnast** [comment]
Tested with `/ihme/homes/pnast/scratch/vivarium/vivarium_gates_nutrition_optimization_child/backup_testing/pakistan/2024_09_11_14_46_24`
in similar manner to #233 ; ran one job to completion, ran 3 more jobs, killing partway after backup creation, observed loading from saved backup.

File: tests/psimulate/worker/test_vivarium_work_horse.py [hunk 1]
```diff
@@ -102,7 +54,13 @@ def test_get_backup(
                     "random_seed": [random_seed],
                     "job_id": job_id,
                     "branch_key": ["branch_value"],
-                }
+                },
+                {
```
Comment (patricktnast) [inline tests/psimulate/worker/test_vivarium_work_horse.py:58]: I added a, let's say, test context here, to make sure that we don't pick up a pickle with the same branch configuration but a different draw or seed. I almost submitted a bug where that could happen.

File: src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py [hunk 2]
```diff
@@ -235,11 +195,13 @@ def get_backup(job_parameters: JobParameters) -> Optional[SimulationContext]:
     metadata_path = job_parameters.backup_configuration["backup_metadata_path"]
     try:
         pickle_metadata = pd.read_csv(metadata_path)
-        query_conditions = f"input_draw == {job_parameters.input_draw} & random_seed == {job_parameters.random_seed}"
-
-        # Add branch parameter conditions to the query
-        for k, v in collapse_nested_dict(job_parameters.branch_configuration):
-            query_conditions += f' & `{k}` == "{v}"'
+        format_val = lambda v: v if type(v) in [int, float] else f'"{v}"'
```
Comment (rmudambi) [inline src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py:198]: Isn't this better?
```suggestion
        format_val = lambda v: v if isinstance(v, (int, float)) else f'"{v}"'
```

**Review by rmudambi — APPROVED** [review]

**Review by stevebachmeier — APPROVED** [review]

File: src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py [hunk 3]
```diff
@@ -215,17 +179,13 @@ def format_and_record_details(
 ) -> pd.DataFrame:
     """Add finished simulation details to results and metadata."""
     finished_results_metadata = pd.DataFrame(index=[0])
-    for key, val in collapse_nested_dict(job_parameters.branch_configuration):
-        # Exclude the run_configuration values from branch_configuration
-        # since they are duplicates. Also do not include the additional_seed
-        # value since it is identical to input_draw
+    for key, val in collapse_nested_dict(job_parameters.job_specific):
+        # Do not include the additional_seed value since it is identical to input_draw
```
[the remark this answers is not in the document]
Comment (patricktnast) [reply src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py]: removed

**Pull request merged**